From 9610a3495173d52be32ebaaa440b1a1c1323690a Mon Sep 17 00:00:00 2001
From: Hebbar, Gururaja <gururaja.hebbar@ti.com>
Date: Mon, 26 Sep 2011 17:03:23 +0530
Subject: [PATCH 11/29] ARM:omap:am33xx: McASP hookup for AM335x EVM

This patch adds the machine driver for AM335x EVM in conformance with the new
ASoC framework.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 sound/soc/davinci/davinci-evm.c |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/sound/soc/davinci/davinci-evm.c b/sound/soc/davinci/davinci-evm.c
index b0fcf79..e173922 100644
--- a/sound/soc/davinci/davinci-evm.c
+++ b/sound/soc/davinci/davinci-evm.c
@@ -55,6 +55,9 @@ static int evm_hw_params(struct snd_pcm_substream *substream,
 	else if (machine_is_davinci_da830_evm() ||
 				machine_is_davinci_da850_evm())
 		sysclk = 24576000;
+	/* On AM335X, CODEC gets MCLK from external Xtal (12MHz). */
+	else if (machine_is_am335xevm())
+		sysclk = 12000000;
 
 	else
 		return -EINVAL;
@@ -240,6 +243,17 @@ static struct snd_soc_dai_link da850_evm_dai = {
 	.ops = &evm_ops,
 };
 
+static struct snd_soc_dai_link am335x_evm_dai = {
+	.name = "TLV320AIC3X",
+	.stream_name = "AIC3X",
+	.cpu_dai_name = "davinci-mcasp.1",
+	.codec_dai_name = "tlv320aic3x-hifi",
+	.codec_name = "tlv320aic3x-codec.2-001b",
+	.platform_name = "davinci-pcm-audio",
+	.init = evm_aic3x_init,
+	.ops = &evm_ops,
+};
+
 /* davinci dm6446 evm audio machine driver */
 static struct snd_soc_card dm6446_snd_soc_card_evm = {
 	.name = "DaVinci DM6446 EVM",
@@ -280,6 +294,12 @@ static struct snd_soc_card da850_snd_soc_card = {
 	.num_links = 1,
 };
 
+static struct snd_soc_card am335x_snd_soc_card = {
+	.name = "AM335X EVM",
+	.dai_link = &am335x_evm_dai,
+	.num_links = 1,
+};
+
 static struct platform_device *evm_snd_device;
 
 static int __init evm_init(void)
@@ -306,6 +326,9 @@ static int __init evm_init(void)
 	} else if (machine_is_davinci_da850_evm()) {
 		evm_snd_dev_data = &da850_snd_soc_card;
 		index = 0;
+	} else if (machine_is_am335xevm()) {
+		evm_snd_dev_data = &am335x_snd_soc_card;
+		index = 0;
 	} else
 		return -EINVAL;
 
-- 
1.6.6.1

